server2/Arcadia.Assistant/Arcadia.Assistant.Sharepoint/SharepointItemSynchronization.cs

```diff
@@ -0,0 +1,143 @@
+﻿namespace Arcadia.Assistant.Sharepoint
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using CalendarEvent;
+
+    using Employees.Contracts;
+
+    using ExternalStorages.Abstractions;
+
+    using Microsoft.Extensions.Logging;
+
+    public abstract class SharepointItemSynchronization<T>
+    {
+        #region ctor
+
+        protected SharepointItemSynchronization(IExternalStorage externalStorage, ILogger? logger = null)
+        {
+            this.ExternalStorage = externalStorage;
+            this.Logger = logger;
+        }
+
+        #endregion
+
+        #region public interface
+
+        public async Task SynchronizeItem(string calendar, EmployeeMetadata[] departmentEmployes, Dictionary<string, T> values, IEnumerable<StorageItem> storageItemsList, CancellationToken cancellationToken)
```

Comment:
Method should be named `SynchronizeItems`.

Reply:
Done